qa/tasks/nvmeof.py

```diff
@@ -134,34 +132,31 @@ def deploy_nvmeof(self):
         
     def set_gateway_cfg(self):
         log.info('[nvmeof]: running set_gateway_cfg...')
-        gateway_config = self.config.get('gateway_config', {})
-        source_host = gateway_config.get('source')
-        target_host = gateway_config.get('target')
-        if not (source_host and target_host):
-            raise ConfigError('gateway_config requires "source" and "target"')
-        remote = list(self.ctx.cluster.only(source_host).remotes.keys())[0]
-        ip_address = remote.ip_address
-        gateway_name = ""
+        ip_address = self.remote.ip_address
+        gateway_names = []
+        gateway_ips = []
         nvmeof_daemons = self.ctx.daemons.iter_daemons_of_role('nvmeof', cluster=self.cluster_name)
         for daemon in nvmeof_daemons:
-            if ip_address == daemon.remote.ip_address:
-                gateway_name = daemon.name()
+            gateway_names += [daemon.name()]
+            gateway_ips += [daemon.remote.ip_address]
         conf_data = dedent(f"""
-            NVMEOF_GATEWAY_IP_ADDRESS={ip_address}
-            NVMEOF_GATEWAY_NAME={gateway_name}
+            NVMEOF_GATEWAY_IP_ADDRESSES={",".join(gateway_ips)}
+            NVMEOF_GATEWAY_NAMES={",".join(gateway_names)}
+            NVMEOF_DEFAULT_GATEWAY_IP_ADDRESS={ip_address}
             NVMEOF_CLI_IMAGE="quay.io/ceph/nvmeof-cli:{self.cli_image}"
+            NVMEOF_NAMESPACES={self.namespaces_count}
```

Comment:
Make it clear that it's not a list like e.g. `NVMEOF_GATEWAY_IP_ADDRESSES`:
```suggestion
            NVMEOF_NAMESPACES_COUNT={self.namespaces_count}
```